### PR TITLE
chore: remove some uses of Batteries.HashMap

### PR DESCRIPTION
**Pull request by kim-em**
Note this is a PR to `nightly-testing`, but can be reviewed as if it is a usual PR.

This switches over some uses of `Batteries.HashMap` in favour of `Std.HashMap`.

We can't switch over the uses in `linarith` yet, because we are missing `mergeWith` and `mapVal`.

### CODE REVIEW

File: Mathlib/Tactic/CC/Addition.lean [hunk 1]
```diff
@@ -1622,25 +1622,25 @@ def removeParents (e : Expr) (parentsToPropagate : Array Expr := #[]) : CCM (Arr
...
+        if let some lst := (← get).symmCongruences[k']? then
           let k := (p, rel)
           let newLst ← lst.filterM fun k₂ => (!·) <$> compareSymm k k₂
           if !newLst.isEmpty then
             modify fun ccs =>
               { ccs with symmCongruences := ccs.symmCongruences.insert k' newLst }
           else
             modify fun ccs =>
-              { ccs with symmCongruences := ccs.symmCongruences.erase k' }
+              { ccs with symmCongruences := ccs.symmCongruences.remove k' }
```
Comment (TwoFX) [inline Mathlib/Tactic/CC/Addition.lean:1633]: leanprover/lean4#4725 is now merged, so this will change back to `erase` with the next nightly.

**Review by TwoFX — APPROVED** [review]
LGTM, but the code will break in the next nightly, so you might want to hold off on merging.
bors d+

**Comment by semorrison** [comment]
bors merge

**Comment by semorrison** [comment]
bors merge

**Pull request closed without merging**